### DISCUSSION TITLE
performance of ndarray arithmetic operation

### DISCUSSION THREAD

**Discussion by tyghl** [rust-ndarray/ndarray#1275] [Q&A] [answered]
i want to perform an Add operation and assign the result to another array.
like this:
```
result.assign(&(&arr1+ &arr2));
```
since i don't want to consume the data in arr1 and arr2, i think this method will always generate a temporary array, hence bad for performance.
So should i use iterator and add seperately in this case? Does any better solution exist?

**Comment by bluss** [comment, marked as answer]
First consider `result += &arr1; result += &arr2;` if you need something custom, you can do it with `Zip` without creating any intermediate array.

**Reply by jturner314** [reply]
Both of those are good suggestions. Using two separate add-assign operations requires two separate loops over the elements, while `Zip`/`azip!` does it in a single loop over the elements. Here's an example with the `azip!` macro:

```rust
use ndarray::azip;
use ndarray::prelude::*;

fn main() {
    let arr1 = array![1, -4, 7];
    let arr2 = array![6, 8, -2];
    let mut result = Array1::zeros(3);
    azip!((r in &mut result, &a1 in &arr1, &a2 in &arr2) *r = a1 + a2);
    println!("{}", result);
}
```

([Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=99003c11c198da7b584c6cae83343ffe))

**Reply by tyghl** [reply]
Thank you very much! `Zip` is exactly what i want!
